File: src/main/java/historywowa/domain/openai/dto/image/req/OpenAIVisionReq.java
package historywowa.domain.openai.dto.image.req;

import java.util.List;

public record OpenAIVisionReq(
        String model,
        List<Message> messages
) {
    public static record Message(
            String role,
            List<Content> content
    ) {}

    public static record Content(
            String type,
            String text,
            ImageUrl image_url
    ) {}

    public static record ImageUrl(
            String url
    ) {}
}

File: src/main/kotlin/historywowa/domain/openai/dto/image/req/OpenAIVisionReq.kt
package historywowa.domain.openai.dto.image.req

data class OpenAIVisionReq(
        val model: String,
        val messages: List<Message>
) {
    data class Message(
            val role: String,
            val content: List<Content>
    )

    sealed interface Content

    data class TextContent(
            val type: String,
            val text: String
    ) : Content

    data class ImageContent(
            val type: String,
            val image_url: ImageUrl
    ) : Content

    data class ImageUrl(
            val url: String
    )
}

